API/Controllers/LibraryController.cs

```diff
@@ -227,6 +227,11 @@ public async Task<ActionResult<IEnumerable<SearchResultDto>>> Search(string quer
         {
             queryString = queryString.Replace(@"%", "");
 
+            if (char.IsWhiteSpace(queryString, 0) || char.IsWhiteSpace(queryString, queryString.Length - 1))
```

Comment:
This isn't very clean. Keep the code simple. Why doesn't just trimming work? Without the checks.